docs/contributor/migration-guide.md

```diff
@@ -0,0 +1,85 @@
+
+3. **Manifest and Default CR data**
+
+   - **Deprecated**: The Manifest and Default CR data cannot be read from a local file.
+   - **New Approach**: The Manifest and Default CR data is fetched directly from the GitHub release. See an example [here](https://github.com/kyma-project/modulectl/blob/91e01856b944fda0d5595843e040bca26416abdc/tests/e2e/create/testdata/moduleconfig/valid/with-defaultcr.yaml#L3-L4).
+
+4. **Annotations and Labels**
+
+   - **Deprecated**: The documentation link is not added as an annotation `operator.kyma-project.io/doc-url`.
+   - **New**: The documentation link is added as **.spec.info.documentation**. The value is configured in the module config file with key **.documentation**.
+   - **Deprecated**: The module version is not added as a label `operator.kyma-project.io/module-version`.
+   - **New**: The module version is added as **.spec.version**.
+
+5. **Command Flags**
+
+   - **Deprecated**: Flag `--module-config-file` is deprecated.
+   - **New**: Flag `--config-file` with shortcut `-c` (applicable for both scaffold and create commands).
+   - **Deprecated**: Flag `--module-archive-version-overwrite` is deprecated.
+     - There is no successor, this feature has been sunset entirely. Reason is that module versions should be immutable once built and pushed. If a version really needs to be re-written, it should be deleted from the registry explicitly first.
+
+6. **ModuleTemplate Naming Pattern**
+
+   - **Deprecated**: **.metadata.name** is not written as `<module-name>-<channel>`.
+   - **New**: **.metadata.name** is written as `<module-name>-<version>`.
+
+7. **Mandatory ModuleTemplates**
+
+   - **Restriction**: At the moment, `modulectl` is not used for the creation of mandatory ModuleTemplates. Please use Kyma CLI instead until the next release.
+
+8. **Beta and Internal Flags**
+
+- **Deprecated**: Beta and Internal flags are not supported for ModuleTemplates.
+- **New**: Beta and Intenral flags are configured as part of the ModuleReleaseMeta, see [ModuleReleaseMeta Configuration](https://github.com/kyma-project/lifecycle-manager/blob/main/docs/contributor/resources/05-modulereleasemeta.md#configuration).
+
+## Migration Period
+
+**Support for Both Approaches Temporarily**: 
+
+- Both old and new approaches will be supported simultaneously by KLM during the migration period.
+- After the full migration to the new approach, KLM will no longer accept ModuleTemplate with the old naming pattern.
+- For testing, provide ModuleTemplates in the new format accompanied by [ModuleReleaseMeta](https://github.com/kyma-project/lifecycle-manager/blob/main/docs/contributor/resources/05-modulereleasemeta.md). KLM will attempt to use the new approach and fall back to the old approach if the new format is not found. To avoid syncing the new ModuleTemplate to the SKR while testing, use the internal label.
```

Comment:
```suggestion
- For testing, provide ModuleTemplate CRs in the new format accompanied by [ModuleReleaseMeta CRs](https://github.com/kyma-project/lifecycle-manager/blob/main/docs/contributor/resources/05-modulereleasemeta.md). KLM will attempt to use the new approach and fall back to the old approach if the new format is not found. Use the internal label, to avoid syncing the new ModuleTemplate to the SKR while testing.
```